src/network.rs

```diff
@@ -0,0 +1,25 @@
+use std;
+use std::sync::mpsc::channel;
+
+use hyper::{Client, Uri};
+use hyper::rt::{self, Future, Stream};
+
+pub fn fetch_http_code_sync(url: Uri) -> std::io::Result<String> {
```

Comment:
Should add a comment here about the purpose of sync http fetch. Something like
"The CodeFetch message is used to load HTTP javascript resources and expects a synchronous response, this utility method supports that."
Rename to "http_code_fetch" ?